src/app/components/Table/Table.tsx

```diff
@@ -99,7 +99,7 @@ export const Table: FunctionComponent<TableProps> = ({
         <Tbody>
           {transformRows(rows, columns).map((row: TableRow, rowIndex) => (
             <Tr
-              ouiaId={row.originalData?.id as string}
+              ouiaId={row.originalData?.name as string}
```

Comment:
It's not strictly related to this PR because the `id` was already used here before, but I'm not sure if it's a good idea to retrieve information from the original data. 

We are inside a generic Table component that shouldn't know what is the type of content rendered in the rows.

Probably it makes more sense to only use the `rowIndex` for both `ouiaId` and `key`, like `row_{rowIndex}`.

What do you think? Also asking @vpellegrino opinion on this.

Reply:
Hi @kelvah
Thanks for asking
I originally added the usage of id for the react key property.
The reason was that all domain models have an id.
By the way I put on the key a fallback on row index.
I don't have a strong opinion. Your consideration about agnosticism from original data is totally fine.

Reply:
@kelvah do you plan to use `Table.tsx` in filters? When do we start using ouiaId than such table's rows usually contain `id` or `name` as ouiaId. See https://github.com/kiegroup/kogito-apps/blob/main/ui-packages/packages/management-console/cypress/integration/processes/page-initial-load.ts#L78-L79

For me it is more readable and safer `cy.ouiaId("New instance", "PF4/TableRow").should("be.visible")` than `cy.ouiaType("PF4/TableRow").contains("New instance").should("be.visible")`. I have not got any performance comparison of these approaches. Both approaches are possible. The first approach compare the whole text and the name should be unique (Processor and Instance creation forms check it). The second one has a problem with "New instance one" and "New instance two". 

I have checked that you are using this table only with Processors and Instances. Both contains `id` and `name`.
![image](https://user-images.githubusercontent.com/1536609/169384608-1acd1443-b5ab-4098-a4fd-0f9ea4a3cfba.png)

When I create new Instance I know only the `name` (not `id`) and I am not sure if the order is stable. Do you mean that  `cy.ouiaId("index-0", "PF4/TableRow").should("have.text","New instance")` will be allways true?

You have totally rigth that curently we are sure that it is not a problem. But we should decide which data construction are applicable in  the `Table` component. Can we require `id` or `name` as mandatory fields? IMHO it is not a scope of this JIRA.

Reply:
> When I create new Instance I know only the `name` (not `id`) and I am not sure if the order is stable. Do you mean that `cy.ouiaId("index-0", "PF4/TableRow").find("td:first").should("have.text","New instance")` will be allways true?

It has to be true. Instances are sorted by default from the latest one. So the test could also be responsible to check that the default sorting order is working properly. And if we use the index as row key / ouiaId we could also avoid using `td:first`.

> 
> You have totally rigth that curently we are sure that it is not a problem. But we should decide which data construction are applicable in the `Table` component. Can we require `id` or `name` as mandatory fields? IMHO it is not a scope of this JIRA.

I agree about the scope, I mentioned it in my first comment. 

Things are technically working with the current solution. Like you pointed out, we have ids and names on both tables data. My comment was more about discussing if it's the right thing to do from a conceptual point of view.
The table should be agnostic about the rows content because it's a generic table. It's not an InstancesTable or ProcessorsTable.

Anyway, we can improve this in a separate ticket. I'm going to approve.